libs/community/langchain_community/graph_vectorstores/extractors/hierarchy_link_extractor.py

```diff
@@ -0,0 +1,74 @@
+from typing import Callable, List, Set
+
+from langchain_core.documents import Document
+from langchain_core.graph_vectorstores.links import Link
+
+from langchain_community.graph_vectorstores.extractors.link_extractor import (
+    LinkExtractor,
+)
+from langchain_community.graph_vectorstores.extractors.link_extractor_adapter import (
+    LinkExtractorAdapter,
+)
+
+# TypeAlias is not available in Python 2.9, we can't use that or the newer `type`.
+HierarchyInput = List[str]
+
+
+class HierarchyLinkExtractor(LinkExtractor[HierarchyInput]):
+    def __init__(
+        self,
+        kind: str = "hierarchy",
+        up_links: bool = True,
+        down_links: bool = False,
+        sibling_links: bool = False,
+    ):
+        """Extract links from a document hierarchy.
```

Comment:
Could you include an example to explain conceptually what the up / down and sibling links are? 

An example would help users to understand things conceptually without reading the implementation.

You have good example in the test case, so could inline 1 or 2 of those examples?